src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpAgoraDAO.scala

```diff
@@ -26,4 +27,15 @@ class HttpAgoraDAO(agoraUrl: String)(implicit val system: ActorSystem, implicit
   override def postNamespacePermissions(ns: String, entity: String, perms: List[AgoraPermission])(implicit userInfo: UserInfo): Future[List[AgoraPermission]] =
     authedRequestToObject[List[AgoraPermission]]( Post(getNamespaceUrl(ns, entity), perms) )
 
+  def status: Future[(Boolean, Option[String])] = {
+    val response = unAuthedRequestToObject[AgoraStatus](Get(Uri(agoraBaseUrl).withPath(Uri.Path("/status"))))
+
+    response map { response =>
+      response.status match {
+        case "up" => (true, None)
+        case "down" => (false, if (response.message.length > 0) Some(response.message.mkString("\r\n")) else None)
```

Comment:
Probably unlikely, but you might want to handle `case _` with a custom error message to catch anything else.